pkg/asset/installconfig/vsphere/permissions.go

```diff
@@ -0,0 +1,378 @@
+		}
+		if len(computeClusters) == 1 {
+			return computeClusters[0], nil
+		}
+		return nil, errors.New("multiple clusters found but no Cluster defined in platform spec")
+	}
+	for _, cluster := range computeClusters {
+		if cluster.Name() == platform.Cluster {
+			return cluster, nil
+		}
+	}
+	return nil, errors.New("specified cluster not found")
+}
+
+func getDatastoreFromConfig(ctx context.Context, platform *vspheretypes.Platform, finder Finder, datacenter *object.Datacenter) (*object.Datastore, error) {
+	datastores, err := finder.DatastoreList(ctx, formatPath(datacenter.InventoryPath))
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to list datastores")
+	}
+	if platform.DefaultDatastore == "" {
+		// API returns an error when no results, but let's leave this in to be defensive.
+		if len(datastores) == 0 {
+			return nil, errors.New("did not find any datastores")
+		}
+		if len(datastores) == 1 {
+			return datastores[0], nil
+		}
+		return nil, errors.New("multiple datastores found but no defaultDatastore defined in platform spec")
+	}
+	for _, ds := range datastores {
+		if ds.Name() == platform.DefaultDatastore {
+			return ds, nil
+		}
+	}
+	return nil, errors.New("specified defaultDatastore not found")
+}
+
+func getDatacenterFromConfig(ctx context.Context, platform *vspheretypes.Platform, finder Finder) (*object.Datacenter, error) {
+	dataCenters, err := finder.DatacenterList(ctx, root)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to list datacenters")
+	}
+	if platform.Datacenter == "" {
+		// API returns an error when no results, but let's leave this in to be defensive.
+		if len(dataCenters) == 0 {
+			return nil, errors.New("did not find any datacenters")
+		}
+		if len(dataCenters) == 1 {
+			return dataCenters[0], nil
+		}
+		return nil, errors.New("multiple datacenters found but not datacenter defined in platform spec")
+	}
+	for _, dc := range dataCenters {
+		if dc.Name() == platform.Datacenter {
+			return dc, nil
+		}
+	}
+	return nil, errors.New("specified datacenter not found")
+}
+
+func getNetworkFromConfig(ctx context.Context, platform *vspheretypes.Platform, finder Finder, datacenter *object.Datacenter) (*vim25types.ManagedObjectReference, error) {
+	networks, err := finder.NetworkList(ctx, formatPath(datacenter.InventoryPath))
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to list networks")
+	}
+
+	if platform.Network == "" {
+		// API returns an error when no results, but let's leave this in to be defensive.
+		if len(networks) == 0 {
+			return nil, errors.New("did not find any networks")
+		}
+		if len(networks) == 1 {
+			networkRef := networks[0].Reference()
+			return &networkRef, nil
+		}
+		return nil, errors.New("multiple networks found but not network defined in platform spec")
+	}
+	targetNetwork := fmt.Sprintf("%s/network/%s", datacenter.InventoryPath, platform.Network)
+	networkReference, err := finder.Network(ctx, targetNetwork)
```

Comment:
If user keeps networks within a folder, this will have same issue as in https://bugzilla.redhat.com/show_bug.cgi?id=2063829.